### DISCUSSION TITLE
How do I customize the Quill editor config that's hardcoded?

### DISCUSSION THREAD

**Discussion by rgozdzialski** [austintoddj/canvas#1160] [Troubleshooting] [answered]
Canvas is using the Quill rich text editor which offers various cool configuration options. This config is hardcoded in Canvas and resides within the `vendor` directory, outside of the Laravel project's "reach".

The config is placed in the editor's Vue file and looks like this:

```
let quill = new Quill(this.$refs.editor, {
                modules: {
                    syntax: true,
                    toolbar: [
                        ['bold', 'italic', 'code', 'link'],
                        [{ header: '2' }, { header: '3' }],
                        ['blockquote', 'code-block'],
                    ],
                },
                theme: 'bubble',
                scrollingContainer: 'html, body',
                placeholder: this.trans.tell_your_story,
            });
```

**How can I override this?**

**Comment by austintoddj** [comment, marked as answer]
It's not customizable at the moment. Working on v7, I have considered multiple editor options. It's still heavily under development, but your point here is a good one. I'll see what it would look like to allow user-editable options.

**Reply by rgozdzialski** [reply]
Got it. And thanks for the nice package!

**Comment by tvarga94** [comment]
@rgozdzialski is there any update on that? Its very strange to me that the editor is that limited. I think alignment should be a very basic functionality when its comes to blogging and we don't even have that...I would definitely would like to add that if possible.

**Comment by austintoddj** [comment]
Closing this out with [Canvas v7.0.0](https://github.com/austintoddj/canvas/releases/tag/v7.0.0).

Quill is gone — the editor is TipTap now, so overriding a hardcoded Quill config no longer applies. Editor behavior is package-owned again; customization is via extending the host app / frontend, not swapping Quill modules in `vendor`.

See the [upgrade guide](https://github.com/austintoddj/canvas/blob/main/.github/UPGRADE.md) for the v7 rewrite.
